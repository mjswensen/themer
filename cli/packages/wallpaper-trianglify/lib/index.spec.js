const { colors } = require('../../colors-default'),
  { render, renderInstructions } = require('./index');

describe('themer trianglify wallpaper', () => {
  it(`should return 8 PNG files to write`, async () => {
    const files = await Promise.all(render(colors, {}));
    expect(files.length).toBe(8);
    expect(files.filter(file => /\.png/.test(file.name)).length).toBe(8);
  });
  it('should list output files', async () => {
    const files = await Promise.all(
      render(
        colors,
        {
          'themer-wallpaper-trianglify-size': '1000x1000',
          'themer-wallpaper-trianglify-variance': '0.5',
        },
      )
    );
    const instructions = renderInstructions(files.map(({ name }) => name));
    expect(instructions).toMatchSnapshot();
  });
});
