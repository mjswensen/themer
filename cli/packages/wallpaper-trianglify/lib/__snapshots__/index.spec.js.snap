// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`themer trianglify wallpaper should list output files 1`] = `
"
Files generated:

* \`themer-wallpaper-trianglify-dark-1000x1000-0.5-1.png\`
* \`themer-wallpaper-trianglify-dark-1000x1000-0.5-2.png\`
* \`themer-wallpaper-trianglify-light-1000x1000-0.5-1.png\`
* \`themer-wallpaper-trianglify-light-1000x1000-0.5-2.png\`
"
`;
